### DISCUSSION TITLE
convert Grayscale images to sRGB

### DISCUSSION THREAD

**Discussion by ronzeiller** [ImageMagick/ImageMagick#2073] [Help]
Hello,
working with Version: ImageMagick 7.0.10-14 Q16 x86_64

Input: 16-bit tiff images with embedded ICC profile "ISO Coated v2 - GREY 1c"

I want to convert to sRGB jpg with embedded profile.

At the moment the command which is working (but not embedding the profile) is:
`convert in_bw.jpg  -verbose -set colorspace GRAY -type truecolor -set colorspace sRGB out_rgb.jpg`

Any suggestions?
Thanks

**Comment by snibgo** [comment]
For v7, I suggest you use "magick", not "convert" or "magick convert".

If your input is one-channel grayscale, with a grayscale ICC, why are your trying to save it as a 3-channel image with a one-channel ICC profile?

Please post your input and output files.

**Comment by ronzeiller** [comment]
Thank you for your reply.

The whole thing is used in Resourcespace DMA.

In this case the input tif files to archive are in an 1c printing profile.
Downloads in larger sizes are done in resized greyscale jpgs with the original ICC profile embedded, which is working fine.

1. conversion in Resourcespace:
`magick grey.tif[0] -flatten -quality 90 -alpha Off lpr.jpg`

For smaller downloads and screen images they should be converted into sRGB.
(If not in sRGB the 1c images with this special printing profile are looking too dark)

2. conversion in RS:
`magick lpr.jpg[0] -flatten -quality 90 -alpha Off -strip -colorspace sRGB -resize 200x133 scr.jpg`

identify scr.jpg gives
scr.jpg JPEG 199x133 199x133+0+0 8-bit Gray 256c 9334B 0.000u 0:00.000
 
The same commands with 4c (CMYK) tif images as images to archive are working fine and as expected. They will be converted to sRGB.

To the attached files:
1. zip is 16bit tif.
2. lpr.jpg is 1c with the original profile embedded. It is looking fine, because color management is working.
3. scr.jpg is also 1c, but as it has no profile embedded it is looking too dark.
On the other hand to embed a profile into a 9kB image is not so nice.
This is why it should be in sRGB which should be looking fine as the program expects more or less sRGB.

So I changed the title of the post, as the conversion to sRGB is needed only.
And, if possible with auto-recognizing the embedded input profile.

[grey.tif.zip](https://github.com/ImageMagick/ImageMagick/files/4701756/grey.tif.zip)
![lpr](https://user-images.githubusercontent.com/12698891/83259453-5b894a80-a1b8-11ea-8ae2-f02adad41b27.jpg)
![scr](https://user-images.githubusercontent.com/12698891/83259458-5deba480-a1b8-11ea-8d3a-c06a92f38f01.jpg)

**Comment by snibgo** [comment]
Your gray.tif contains an interesting profile. It is single channel, with a transfer response curve and nothing else. The TRC is similar to sRGB: zero is close to black, and 100% is white, but it generally has significantly lower values than sRGB. I suppose the ICC is designed for a particular printer.

Hence, software that ignores the profile, and assumes the data is sRGB, will show a darker image than software that correctly processes the image.

To correctly convert to sRGB:

`magick gray.tif -profile sRGB.icc -strip out.jpg`

Strictly, you shouldn't strip the profile, because sRGB isn't a solid standard. But stripping will save space, and won't make much difference.

However, you might have been viewing the image with software that ignores the profile, and that's the image you want as sRGB. In that case, simply strip the profile:

`magick gray.tif -strip out2.jpg`

This result will be darker (but wrong).

I suggest you try both methods.

**Comment by ronzeiller** [comment]
Thank you.

>To correctly convert to sRGB:
> `magick gray.tif -profile sRGB.icc -strip out.jpg`

If I do this I still get a greyscale image. (attached)
identify out-srgb-im.jpg 
JPEG 300x200 300x200+0+0 8-bit Gray 256c 20511B 0.010u 0:00.000

If I do `magick gray.tif -profile sRGB.icc out.jpg`
Photoshop claims, that the embedded profile is wrong. (sRGB profile embedded in greyscale image)

Best solution up to now is:
`magick  grey.tif -profile ISOcoated_v2_sw_bas.ICC -profile sRGB.icc -resize 300x200 out_iccgray-srgb-im-strip.jpg`
This results in an sRGB jpg with no profile embedded. (attached)
I agree, that it would be better to always embed profile  for a correct look if opened in Photoshop and other color managed environments. May be not in thumbnails, but at least in the little bit larger previews.

It is a little bit disappointing, that it seems that IM cannot convert from greyscale to RGB by the embedded profile.

![out-srgb-im](https://user-images.githubusercontent.com/12698891/83274776-82eb1200-a1ce-11ea-94b4-3624405d4601.jpg)

![out_iccgray-srgb-im-strip](https://user-images.githubusercontent.com/12698891/83276199-90090080-a1d0-11ea-9fec-960fbdcc14cf.jpg)

**Comment by snibgo** [comment]
It can.

`magick grey.tif -profile sRGB.icc -strip -profile sRGB.icc -type truecolor out.jpg`

This converts to sRGB. Then we "-strip" to remove Photoshop etc metadata, and re-add the correct profile that has been stripped. "-type truecolor" gives us three channels.

Is that what you want?

**Comment by ronzeiller** [comment]
I am afraid not.
magick grey.tif -profile sRGB.icc -strip -profile sRGB.icc -type truecolor out.jpg
probably assumes a gamma of 2.2 (as sRGB) for the input.

above method brings the left image, opened the 1c image the right one:

![screenshot-photoshop](https://user-images.githubusercontent.com/12698891/83284966-c0a36700-a1dd-11ea-80dc-297f7fc29014.png)

**Comment by ronzeiller** [comment]
Some tests showed, that following is working:

`magick in_grey.tif[0] -quiet -depth 8 -flatten  -profile ISOcoated_v2_sw_bas.ICC -intent perceptual -black-point-compensation -profile sRGB_IEC61966-2-1_black_scaled.icc -quality 90 -type truecolor  out_sRGB.jpg`

It converts a 1c-image (even 16 bit) into sRGB and embeds the RGB ICC profile.
If the source profile is not known, someone has to extract it, e.g. by 
`magick imagename.extension icc-profile.icc`
